kotlinpoet/src/nonJvmMain/kotlin/com/squareup/kotlinpoet/Util.nonJvm.kt

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (C) 2024 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.kotlinpoet
+
+internal actual fun <K, V> Map<K, V>.toImmutableMap(): Map<K, V> =
+  toMap()
+
+internal actual fun <T> Collection<T>.toImmutableList(): List<T> =
+  toList()
+
+internal actual fun <T> Collection<T>.toImmutableSet(): Set<T> =
+  toSet()
```

Comment:
If we're not going to put these in an unmodifiable wrapper on other targets, do we still need to do it on JVM? Are we really worried about Java callers mutating things? I don't think I am.

Reply:
The original implementation was copied when we forked from JavaPoet, for what it's worth.

Reply:
Well, you have a point. If we no longer care about calls that mutate things on the JVM, do we keep `toImmutable*` in common and have its result just return a `to*` (like `toSet`), or do we simply remove them and use `to*` instead where appropriate

Reply:
I think it's defending against something that is not a real problem in practice, and I would eliminate the concept entirely and use Kotlin's collection APIs directly everywhere. This should probably be done in its own PR rather than cluttering this one.